kong/runloop/handler.lua

```diff
@@ -786,6 +791,9 @@ do
       router_version = version
     end
 
+    router_cache:flush_all()
+    router_cache_neg:flush_all()
```

Comment:
It does not matter as there is no yielding. They should not be flushed right before because the router build might yield. And then the old router gets used, and now we have destroyed cache for old router. I think it is fine as is. It could be right after, but definitely not right before.